mne/io/eeglab/eeglab.py

```diff
@@ -87,49 +110,92 @@ def _get_info(eeg, montage, eog=()):
+                pos.append(locs)
+    n_channels_with_pos = len(pos_ch_names)
+    if n_channels_with_pos > 0:
+        selection = np.arange(n_channels_with_pos)
+        montage = Montage(np.array(pos), pos_ch_names, kind, selection)
+    else:
+        montage = None
+
+    return ch_names, montage, update_ch_names
+
+
+def _get_info_no_montage(eeg, montage, eog=()):
+    eeg_montage = None
+    update_ch_names = True
+
+    eeg_has_ch_names_info = len(eeg.chanlocs) > 0
+
+    if eeg_has_ch_names_info:
+        has_pos = _eeg_has_montage_information(eeg)
+
+    if eeg_has_ch_names_info:
+        # assert False
         get_pos = has_pos and montage is None
-        pos_ch_names, ch_names, pos = list(), list(), list()
-        kind = 'user_defined'
-        update_ch_names = False
-        for chanloc in eeg.chanlocs:
-            ch_names.append(chanloc['labels'])
-            if get_pos:
-                loc_x = _to_loc(chanloc['X'])
-                loc_y = _to_loc(chanloc['Y'])
-                loc_z = _to_loc(chanloc['Z'])
-                locs = np.r_[-loc_y, loc_x, loc_z]
-                if not np.any(np.isnan(locs)):
-                    pos_ch_names.append(chanloc['labels'])
-                    pos.append(locs)
-        n_channels_with_pos = len(pos_ch_names)
-        info = create_info(ch_names, eeg.srate, ch_types='eeg')
-        if n_channels_with_pos > 0:
-            selection = np.arange(n_channels_with_pos)
-            montage = Montage(np.array(pos), pos_ch_names, kind, selection)
-    elif isinstance(montage, str):
-        path = op.dirname(montage)
+        ch_names, eeg_montage, update_ch_names = _get_eeg_montage_information(
+            eeg, get_pos)
+
     else:  # if eeg.chanlocs is empty, we still need default chan names
         ch_names = ["EEG %03d" % ii for ii in range(eeg.nbchan)]
 
-    if montage is None:
-        info = create_info(ch_names, eeg.srate, ch_types='eeg')
+    return eeg_montage, update_ch_names, ch_names
+
+
+def _get_info(eeg, montage, eog=()):
+    """Get measurement info."""
+    # import pdb; pdb.set_trace()
+
+    eeg = _massage_eeg(eeg)
+
+    # import pdb; pdb.set_trace()
+    eeg_montage, update_ch_names, ch_names = _get_info_no_montage(
+        eeg, montage, eog)
+
+    info = create_info(ch_names, sfreq=eeg.srate, ch_types='eeg')
+
+    if montage is None and eeg_montage is None:
+        pass
     else:
-        _check_update_montage(
-            info, montage, path=path, update_ch_names=update_ch_names,
-            raise_missing=False)
-
-    if eog == 'auto':
-        eog = _find_channels(ch_names)
-
-    for idx, ch in enumerate(info['chs']):
-        ch['cal'] = CAL
-        if ch['ch_name'] in eog or idx in eog:
-            ch['coil_type'] = FIFF.FIFFV_COIL_NONE
-            ch['kind'] = FIFF.FIFFV_EOG_CH
+
+        if eog == 'auto':
+            eog = _find_channels(ch_names)
+
+        for idx, ch in enumerate(info['chs']):
+            ch['cal'] = CAL
```

Comment:
@jasmainak I've a question. Which channels need to be update with cal equal to `1e-6`?
I think it would be better to do a pass at the ned and just look for those that are eeg.

Reply:
@massich was this resolved? do you still need my help here?

Reply:
see #6491